### DISCUSSION TITLE
how to make the previewed UI the same as the compile one?

### DISCUSSION THREAD

**Discussion by ecruzolivera** [slint-ui/slint#4731] [Q&A]
Hi all, I'm learning slint and for some reason the UI that I'm building looks fine in the preview screen, but when running it the dimensions change and seems that the button components are the only ones that keep the correct size.

Does anyone know what could be happening?

Thanks in advance.

This is the repository:

https://github.com/ecruzolivera/vending-machine-collection/tree/main/rs-slint/ui

![image](https://github.com/slint-ui/slint/assets/6627389/511517f7-afa8-4171-b9da-284f1e107f59)

**Comment by hunger** [comment]
It looks like the window is much bigger than the preview.

You could force the size by setting `width` and `height` instead of using `preferred-width` and `preferred-height`. The windowing system is free to pick whatever it thinks best, with the preferred sizes providing a hint.

The style seems to be different between the images you show. That can change the size of buttons, etc.

Another thing that effects sizes is of course the scale-factor.

**Comment by ogoffart** [comment]
This is the difference between the Qt backend and style compared to the winit backend. Qt and winit might compute the scale factor differently. But i'm not sure why this is so different.
You can force in your application the use of a specific backend by specifying that as cargo features.
